tests/src/core/ArrayVectorTest.cpp

```diff
@@ -228,7 +228,7 @@ private slots:
 			// Self-assignment should not change the contents
 			auto v = ArrayVector<int, 5>{1, 2, 3};
 			const auto oldValue = v;
-			v = std::move(v);
+			v = v;
```

Comment:
This defeats the purpose of the test, which is that `ArrayVector` should be protected against self-moves. If you could add a flag to let the compiler ignore self-moves in the unit tests, I think that would be best

Reply:
@messmerd sure, but finally this won't be handled in this PR anyway, I just wanted to see if there are any other points of failure in the build process (there ain't). I'll create a separate ticket for that tomorrow, would
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wself-move"
			v = std::move(v);
#pragma GCC diagnostic pop
```

be OK here? (MSVC doesn't care about it I think)

Reply:
Also, as a side note: self-moves are now prevented on the compiler level, so in all honesty I don't know if this test is useful with new GCC at all (it does have a purpose with current MSVC though IIUC).

Reply:
@messmerd BTW, Linux build in this branch still has GCC too old to recognize that switch (I didn't update it, it's completely besides the scope here). Thus, disabling this W flag won't work unless either both GCCs are bumped simultaneously or unless it's disabled selectively for GCC >= 13. I've went with the latter here for the sake of simplicity and compatibility, although arguably GCC version for Linux build should be bumped as well (even if later on).

Reply:
@messmerd https://github.com/LMMS/lmms/pull/7288

Reply:
> Also, as a side note: self-moves are now prevented in GCC on the compiler level

I'm not sure if that's permitted, and doesn't seem to be the case even with the latest compiler version, full optimisations, and the latest standard - see [an example](https://godbolt.org/z/YMKbd6dKK). Of course, the compiler can optimise it out if it doesn't do anything, but if the compiler can prove that, that's equivalent to the test passing.

> reading through https://www.open-std.org/jtc1/sc22/wg21/docs/lwg-defects.html#2468 shows that writing a test for a case that's unpredictable by itself (due to being highly implementation-dependent and subject to change on the language definition level) is IMVHO moot... *x = move(x) is defined behavior and it leaves x in a valid but unspecified state* is the real problem here: by writing test we say we expect something, but you can't expect anything about an unspecified state, apart from the fact it shouldn't crash the app maybe.

"`x = move(x)` is defined behavior and it leaves `x` in a valid but unspecified state" was written in the context of the specification of standard library containers, not the language itself. Whether it's useful for our class to offer stronger guarantees than the standard library is one question, but the language certainly doesn't prevent it from doing so. The test is useful as long as we do offer that guarantee, and indeed all guarantees should be backed up with a test - perhaps even, without sufficient documentation of the class under test, the tests themselves are what offer the guarantees.

Reply:
@DomClark you're most probably right here TBH, I'm not a C++ expert nor language lawyer, not my area of expertise by a mile. That was just my best guess on the current mixup w.r.t this - and that's just because I read all the discussions concerning it, and not because I have a strong opinion of my own on this matter or because I feel that I understand what all this is really about. That's basically why I went with https://github.com/LMMS/lmms/pull/7288 , because frankly the amount of time this one specific particular compiler warning/error has taken me in LMMS project is now counted in tens of hours, literally. I'd say that's enough for me :D Anyway, I edited my message above, no need to keep my ramblings outlined as they would be holy truth :D
BTW, I'm not a C++ fan mostly due to how obscure and muddy those details are. I'm with you here though - if you have any ideas how to improve this code further I'm all ears.